Sources/UI/Chat/Chat View Controller/ChatViewController.swift

```diff
@@ -51,14 +51,13 @@ open class ChatViewController: ViewController, UITableViewDataSource, UITableVie
     
     /// Calculates the bottom inset for the `ComposerView` when the keyboard will appear.
     open var calculatedSafeAreaBottom: CGFloat {
-        var bottom = view.safeAreaInsets.bottom
-        bottom = bottom > 0 ? bottom : (parent?.view.safeAreaInsets.bottom ?? 0)
-        
-        if let tabBar = tabBarController?.tabBar, !tabBar.isTranslucent {
-            bottom += tabBar.frame.height
+        if let tabBar = tabBarController?.tabBar, !tabBar.isTranslucent, !hidesBottomBarWhenPushed {
+            return tabBar.frame.height
+        } else {
```

Comment:
No need `else` here, was `return` on the previous line.

Reply:
We don't need to access `view.safeAreaInsets` if the view has `tabBarController` so I've moved the logic inside `else`

Reply:
```swift
open var calculatedSafeAreaBottom: CGFloat {
    if let tabBar = tabBarController?.tabBar, !tabBar.isTranslucent, !hidesBottomBarWhenPushed {
        return tabBar.frame.height
    }

    var bottom = view.safeAreaInsets.bottom
    bottom = bottom > 0 ? bottom : (parent?.view.safeAreaInsets.bottom ?? 0)
    return bottom
}
```
Remove `else`

Reply:
and no need `bottom` as `var`